cli/compose/template/template.go

```diff
@@ -176,15 +176,21 @@ func extractVariable(value interface{}, pattern *regexp.Regexp) ([]extractedValu
 
 // Soft default (fall back if unset or empty)
 func softDefault(substitution string, mapping Mapping) (string, bool, error) {
-	return withDefault(substitution, mapping, "-:")
+	sep := ":-"
+	if !strings.Contains(substitution, sep) {
```

Comment:
I guess it can be factorized with `hardDefault` but I don't see an obvious and/or clean way to do it.

Reply:
Yes, I was looking at that, but adding an (e.g.) `checkIfSet func(val) bool` argument felt like over engineering just to prevent just a few lines of duplicated code.